src/components/app/routes/loaders/updateActiveEnterpriseCustomerUserLoader.js

```diff
@@ -27,37 +28,45 @@ export default function makeUpdateActiveEnterpriseCustomerUserLoader(queryClient
       return null;
     }
 
-    if (nextEnterpriseSlug !== activeEnterpriseCustomer.slug) {
+    if (enterpriseSlug !== activeEnterpriseCustomer.slug) {
       // Otherwise, try to find the enterprise customer for the given slug and update it as the active
       // enterprise customer for the learner.
-      const foundEnterpriseCustomerForSlug = allLinkedEnterpriseCustomerUsers.find(
-        enterpriseCustomerUser => enterpriseCustomerUser.enterpriseCustomer.slug === nextEnterpriseSlug,
+      const foundEnterpriseCustomerUserForSlug = allLinkedEnterpriseCustomerUsers.find(
+        enterpriseCustomerUser => enterpriseCustomerUser.enterpriseCustomer.slug === enterpriseSlug,
       );
-      if (foundEnterpriseCustomerForSlug) {
+      if (foundEnterpriseCustomerUserForSlug) {
         await updateUserActiveEnterprise({
-          enterpriseCustomer: foundEnterpriseCustomerForSlug.enterpriseCustomer,
+          enterpriseCustomer: foundEnterpriseCustomerUserForSlug.enterpriseCustomer,
         });
         queryClient.setQueryData(linkedEnterpriseCustomersQuery.queryKey, {
           ...enterpriseLearnerData,
-          activeEnterpriseCustomer: foundEnterpriseCustomerForSlug.enterpriseCustomer,
+          activeEnterpriseCustomer: foundEnterpriseCustomerUserForSlug.enterpriseCustomer,
           allLinkedEnterpriseCustomerUsers: allLinkedEnterpriseCustomerUsers.map(
             ecu => ({
               ...ecu,
               active: (
-                ecu.enterpriseCustomer.uuid === foundEnterpriseCustomerForSlug.enterpriseCustomer.uuid
+                ecu.enterpriseCustomer.uuid === foundEnterpriseCustomerUserForSlug.enterpriseCustomer.uuid
               ),
             }),
           ),
         });
         await Promise.all(getEnterpriseAppData({
-          enterpriseCustomer: foundEnterpriseCustomerForSlug.enterpriseCustomer,
+          enterpriseCustomer: foundEnterpriseCustomerUserForSlug.enterpriseCustomer,
           userId,
           userEmail,
           queryClient,
         }));
         return null;
       }
 
+      const nextEnterpriseLearnerQuery = makeEnterpriseLearnerQuery(username, activeEnterpriseCustomer.slug);
```

Comment:
[inform] Performs an optimistic query cache update to avoid duplicate API `enterprise-learner` call once the app redirects to the slug on line 70 below, given we have all the same data it will already return anyway; the only difference is the query key first included `null` for `enterpriseSlug`, but after the redirect the query key includes a real slug. By setting the existing API response data for the query key _with_ the slug, it stays fresh upon the redirect route transition.